google/cloud/pubsub/internal/publisher_tracing_connection_test.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/publisher_tracing_connection.h"
+#include "google/cloud/pubsub/mocks/mock_publisher_connection.h"
+#include "google/cloud/pubsub/publisher_connection.h"
+#include "google/cloud/pubsub/topic.h"
+#include "google/cloud/internal/opentelemetry.h"
+#include "google/cloud/opentelemetry_options.h"
+#include "google/cloud/testing_util/opentelemetry_matchers.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+namespace {
+
+using ::google::cloud::pubsub::PublisherConnection;
+using ::google::cloud::pubsub::Topic;
+using ::google::cloud::pubsub_internal::PublisherTracingConnection;
+using ::google::cloud::pubsub_mocks::MockPublisherConnection;
+using ::google::cloud::testing_util::InstallSpanCatcher;
+using ::google::cloud::testing_util::SpanCatcher;
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
```

Comment:
```suggestion
#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
using ::google::cloud::pubsub::PublisherConnection;
using ::google::cloud::pubsub::Topic;
using ::google::cloud::pubsub_internal::PublisherTracingConnection;
using ::google::cloud::pubsub_mocks::MockPublisherConnection;
using ::google::cloud::testing_util::InstallSpanCatcher;
using ::google::cloud::testing_util::SpanCatcher;
```

You should probably run `otel-disabled-bazel-pr` locally, a lot of the builds seem to be failing because the code assumes OTel is enabled.